src/PrestaShopBundle/Bridge/Smarty/SmartyTrait.php

```diff
@@ -49,211 +47,11 @@ trait SmartyTrait
      */
     public function renderSmarty(string $content, Response $response = null, bool $isNewTheme = false): Response
     {
-        $this->setMedia($isNewTheme);
+        $this->legacyControllerBridge->setMedia($isNewTheme);
```

Comment:
```suggestion
        $this->getLegacyControllerBridge()->setMedia($isNewTheme);
```
If you go with the suggestion I made about the getter implementation, we could use it here which guarantees a bit more the presence of the object. At least it has more chances to work even if the listener hasn't been called yet.